### DISCUSSION TITLE
I want to export each line of text in pdf to both png and txt files separately, any code/method to do this? steve8000818@gmail.com

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2026] [Q&A] [answered]
I want to export each line of text in pdf to both png and txt files separately, any code/method to do this? steve8000818@gmail.com

**Comment by JorjMcKie** [comment, marked as answer]
Use a variant of text extraction that delivers on line level together with position information.
Then make a pixmap of the line boundary box to output as png:
```python
for block in page.get_text("dict", flags=fitz.TEXTFLAGS_TEXT)["blocks"]:
    for line in block["lines"]:
        bbox = line["bbox"]  # the line bbox
        text = " ".join([span["text"] for span in line["spans"]])  # text in line
        pix = page.get_pixmap(clip=bbox)  # pixmap of line bbox
        pix.save(...)
```

**Comment by nissansz** [comment]
I can save to png, but the text in png is very small, how to enlarge to a height of 32pixels etc.?

**Reply by JorjMcKie** [reply]
use the dpi parameter or the matrix parameter of the `get_pixmap()` method.

**Comment by nissansz** [comment]
pix = page.get_pixmap(clip=bbox, dpi=300)  # pixmap of line bbox
It works.
